File: tdd/calculator/calculator.py
history = []

def add(a, b):
    result = a + b
    history.append(f"{a} + {b} = {result}")
    return result

def subtract(a, b):
    result = a - b
    history.append(f"{a} - {b} = {result}")
    return result

def multiply(a, b):
    result = a * b
    history.append(f"{a} * {b} = {result}")
    return result

def divide(a, b):
    if b == 0:
        raise ValueError("Cannot divide by zero")
    result = a / b
    history.append(f"{a} / {b} = {result}")
    return result

def power(a, b):
    result = a ** b
    history.append(f"{a} ^ {b} = {result}")
    return result

def sqrt(a):
    if a < 0:
        raise ValueError("Cannot calculate the square root of a negative number")
    result = a ** 0.5
    history.append(f"√{a} = {result}")
    return result

def get_history():
    return history

def clear_history():
    global history
    history = []


if __name__ == "__main__":
    # Example usage
    print(add(1, 2))  # Output: 3
    print(subtract(5, 3))  # Output: 2
    print(multiply(2, 4))  # Output: 8
    print(divide(8, 2))  # Output: 4.0
    print(power(2, 3))  # Output: 8
    print(sqrt(16))  # Output: 4.0

    print(get_history())  # Output: history of calculations

File: projeto-testes/src/main/java/com/projeto_tdd/projeto/testes/Calculator.java
package com.projeto_tdd.projeto.testes;

import java.util.ArrayList;
import java.util.List;

public class Calculator {
    private List<String> history = new ArrayList<>();

    public double add(double a, double b) {
        double result = a + b;
        history.add(a + " + " + b + " = " + result);
        return result;
    }

    public double subtract(double a, double b) {
        double result = a - b;
        history.add(a + " - " + b + " = " + result);
        return result;
    }

    public double multiply(double a, double b) {
        double result = a * b;
        history.add(a + " * " + b + " = " + result);
        return result;
    }

    public double divide(double a, double b) throws ArithmeticException {
        if (b == 0) {
            throw new ArithmeticException("Cannot divide by zero");
        }
        double result = a / b;
        history.add(a + " / " + b + " = " + result);
        return result;
    }

    public double power(double a, double b) {
        double result = Math.pow(a, b);
        history.add(a + " ^ " + b + " = " + result);
        return result;
    }

    public double sqrt(double a) {
        if (a < 0) {
            throw new ArithmeticException("Cannot calculate the square root of a negative number");
        }
        double result = Math.sqrt(a);
        history.add("√" + a + " = " + result);
        return result;
    }

    public List<String> getHistory() {
        return history;
    }

    public void clearHistory() {
        history.clear();
    }
}
